examples/SimpleExample/main.cpp

```diff
@@ -68,17 +68,18 @@ int main(int argc, char* argv[])
 
     mechaspin::parakeet::Driver parakeetSensorDriver;
 
-    std::string comPort = argv[1];
-    mechaspin::parakeet::BaudRate baudRate(atoi(argv[2]));
-    bool useDataSmoothing = false;
-    bool useDragPointRemoval = false;
-    bool intensity = true;
-    mechaspin::parakeet::Driver::ScanningFrequency startingScanFrequency_Hz = mechaspin::parakeet::Driver::ScanningFrequency::Frequency_10Hz;
+    mechaspin::parakeet::Driver::SensorConfiguration sensorConfiguration;
+    sensorConfiguration.comPort = argv[1];
+    sensorConfiguration.baudRate = mechaspin::parakeet::BaudRate(atoi(argv[2]));
+    sensorConfiguration.dataSmoothing = false;
+    sensorConfiguration.dragPointRemoval = false;
+    sensorConfiguration.intensity = true;
+    sensorConfiguration.scanningFrequency_Hz = mechaspin::parakeet::Driver::ScanningFrequency::Frequency_10Hz;
 
     std::cout << "Attempting connection to sensor." << std::endl;
-    if(!parakeetSensorDriver.connect(comPort, baudRate, intensity, startingScanFrequency_Hz, useDataSmoothing, useDragPointRemoval))
+    if(!parakeetSensorDriver.connect(sensorConfiguration))
```

Comment:
I'm wondering if connect() should thrown an Exception instead of return true/false. What happens if a user doesn't check that the connect() succeeded? Will the application continue happily? They won't receive any data and be confused? 

Since we added the new connect(SensorConfiguration) method and deprecated the old version, it seems like we could just make the new version through the exception?

Reply:
If the connection fails and the user tries to call any other functions on the Driver, it prints out that the Driver wasn't connected

I believe exceptions should be thrown if an action occurs that relies on the sensor being connected, but isn't. Doing so would affect many functions.

I'm personally more of a fan of connect returning true/false than throwing an exception. The difference is that the exception is forced to be handled to continue. I see pros and cons of both.